code/_onclick/click.dm

```diff
@@ -270,17 +270,19 @@
 	return
 
 /atom/proc/AltClick(var/mob/user)
-	if(!(user == src) && !(isrobot(user)) && iscarbon(src) && user.Adjacent(src))
-		src.give_item(user)
-		return
 	var/turf/T = get_turf(src)
 	if(T && T.Adjacent(user))
 		if(user.listed_turf == T)
 			user.listed_turf = null
 		else
 			user.listed_turf = T
 			user.client.statpanel = T.name
-	return
+
+/mob/living/carbon/AltClick(var/mob/user)
+	if(!(user == src) && !(isrobot(user)) && iscarbon(src) && user.Adjacent(src))
```

Comment:
Since this is a `/mob/living/carbon` proc, `src` is always going to be a carbon.
You don't need the `iscarbon()` check.